packages/fxa-auth-server/lib/senders/select_email_services.js

```diff
@@ -17,6 +16,98 @@ const SERVICES = {
   },
 };
 
+// Live config looks like this (every property is optional):
+//
+// {
+//   sendgrid: {
+//     percentage: 100,
+//     regex: "^.+@example\.com$"
+//   },
+//   socketlabs: {
+//     percentage: 100,
+//     regex: "^.+@example\.org$"
+//   },
+//   ses: {
+//     percentage: 10,
+//     regex: ".*",
+//   }
+// }
+//
+// Where a percentage and a regex are both present, an email address must
+// satisfy both criteria to count as a match. Where an email address matches
+// sendgrid and ses, sendgrid wins. Where an email address matches socketlabs
+// and ses, socketlabs wins. Where an email address matches sendgrid and
+// socketlabs, sendgrid wins.
+//
+// Regular expressions are executed using the `SandboxedRegExp` class in order
+// to protect againt DoS if a malicious regex happened to make its way into the
+// live config; as a result the regex syntax may differ from that of standard
+// javascript, but you shouldn't be putting complex regex in here anyway.
```

Comment:
I'd like a link to https://docs.rs/regex/1.3.6/regex/#syntax or even https://github.com/rfk/sandboxed-regexp#being-careful-with-it here